### PR TITLE
feat(OnyxNavItem): Implement OnyxNavItem

### PR DESCRIPTION
**Pull request by MajaZarkova**
Relates to #883 

Implemented new OnyxNavItem component which is similar to OnyxListItem, but will be used in OnyxNavButton only.

## Checklist

- [x] If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) is added
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

File: packages/sit-onyx/src/components/OnyxNavItem/future/OnyxNavItem.vue [hunk 1]
```diff
@@ -0,0 +1,75 @@
+<script setup lang="ts">
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxNavItem/future/OnyxNavItem.vue:1]: question: is it intentional that the navItems do not yet emit any click event?

File: packages/sit-onyx/src/components/OnyxNavItem/future/OnyxNavItem.vue [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+const props = defineProps<{ active?: boolean }>();
+</script>
+<template>
+  <li
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxNavItem/future/OnyxNavItem.vue:5]: should it really be a li?
in the DOM, we now have li inside li
image

File: packages/sit-onyx/src/components/OnyxNavItem/future/OnyxNavItem.vue [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    :class="{
+      'onyx-future-nav-item': true,
+      'onyx-future-nav-item--active': props.active,
+    }"
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxNavItem/future/OnyxNavItem.vue:9]:
```suggestion
    class="onyx-future-nav-item" 
    :class="{ 'onyx-future-nav-item--active': props.active }"
```

**Review by BoppLi — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/components/OnyxNavButton/OnyxNavButton.stories.ts [hunk 4]
```diff
@@ -59,7 +59,10 @@ export const WithChildren: Story = {
   args: {
     ...Default.args,
     active: true,
-    children: () => nestedChildren.map(({ label }) => h(OnyxListItem, label)),
+    children: () =>
+      nestedChildren.map(({ label, active }) =>
+        h(OnyxNavItem, { "aria-selected": active }, [label]),
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxNavButton/OnyxNavButton.stories.ts:64]: We should only set the aria attribute internally and create a "real" selected/active property :)

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/components/OnyxNavItem/future/OnyxNavItem.vue (hunk 2 above)
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxNavItem/future/OnyxNavItem.vue:5]: Don't we want to use the list item here internally?

File: packages/sit-onyx/src/components/OnyxNavItem/future/OnyxNavItem.vue (hunk 1 above)
Comment (MajaZarkova) [reply packages/sit-onyx/src/components/OnyxNavItem/future/OnyxNavItem.vue:1]: added a click event

File: packages/sit-onyx/src/components/OnyxNavItem/future/OnyxNavItem.vue (hunk 3 above)
Comment (MajaZarkova) [reply packages/sit-onyx/src/components/OnyxNavItem/future/OnyxNavItem.vue]: I refactored the component and it's using the list item internally now. We don't have the classes any more :)

File: packages/sit-onyx/src/components/OnyxNavItem/future/OnyxNavItem.vue (hunk 2 above)
Comment (MajaZarkova) [reply packages/sit-onyx/src/components/OnyxNavItem/future/OnyxNavItem.vue]: I refactored the nav item and we are using the list item internally which means this is not the structure any more :)

File: packages/headless/src/composables/menuButton/createMenuButton.ts [hunk 5]
```diff
@@ -3,14 +3,7 @@ import { createBuilder } from "../../utils/builder";
...
 
-export type CreateMenuButtonOptions = {
-  /**
-   * Called when a menu item is selected (via mouse or keyboard).
-   */
-  onSelect: (value: string) => void;
-};
-
-export const createMenuButton = createBuilder((options: CreateMenuButtonOptions) => {
+export const createMenuButton = createBuilder(() => {
```
Comment (JoCa96) [inline packages/headless/src/composables/menuButton/createMenuButton.ts:6]:
```suggestion
/**
 * Based on https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/examples/disclosure-navigation/
 */
export const createMenuButton = createBuilder(() => {
```

File: packages/headless/src/composables/menuButton/createMenuButton.ts [hunk 6]
```diff
@@ -32,6 +25,68 @@ export const createMenuButton = createBuilder((options: CreateMenuButtonOptions)
...
   });
 
+  const getMenu = (el: HTMLElement) => {
+    const id = el.getAttribute("aria-controls") || "";
+    return document.getElementById(id);
+  };
+
+  const focusRelativeItem = (next: "next" | "prev" | "first" | "last") => {
+    const currentMenuItem = document.activeElement as HTMLElement;
+    const currentMenu = currentMenuItem?.closest('[role="menu"]') || getMenu(currentMenuItem);
```
Comment (JoCa96) [inline packages/headless/src/composables/menuButton/createMenuButton.ts:35]:
```suggestion
    // Either the current focus is on a "menuitem", then we can just get the parent menu.
    // Or the current focus is on the button, then we can get the connected menu using the menuId
    const currentMenu = currentMenuItem?.closest('[role="menu"]') || getMenu(currentMenuItem);
```

File: packages/headless/src/composables/menuButton/createMenuButton.ts [hunk 7]
```diff
@@ -32,6 +25,68 @@ export const createMenuButton = createBuilder((options: CreateMenuButtonOptions)
...
+    if (!currentMenu) return;
```
Comment (JoCa96) [inline packages/headless/src/composables/menuButton/createMenuButton.ts:36]: We don't actually need to get the "aria-controls" value, as we already define the id ourselves:
```suggestion
  const focusRelativeItem = (next: "next" | "prev" | "first" | "last") => {
    const currentMenuItem = document.activeElement as HTMLElement;
    const currentMenu = currentMenuItem?.closest('[role="menu"]') || document.getElementById(menuId);
    if (!currentMenu) return;
```

File: packages/headless/src/composables/menuButton/createMenuButton.ts [hunk 8]
```diff
@@ -3,14 +3,7 @@ import { createBuilder } from "../../utils/builder";
...
-   * Called when a menu item is selected (via mouse or keyboard).
-   */
-  onSelect: (value: string) => void;
-};
-
-export const createMenuButton = createBuilder((options: CreateMenuButtonOptions) => {
+export const createMenuButton = createBuilder(() => {
   const menuId = createId("menu");
   const buttonId = createId("menu-button");
   const isExpanded = ref<boolean>(false);
```
Comment (JoCa96) [inline packages/headless/src/composables/menuButton/createMenuButton.ts:9]: redundant generic:
```suggestion
  const isExpanded = ref(false);
```

File: packages/sit-onyx/src/components/OnyxNavItem/future/types.ts [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+  href?: string;
+  active?: boolean;
+};
+
+/**
+ * [Vue injection key](https://vuejs.org/guide/components/provide-inject) that is provided by the nav bar
+ * to communicate child components whether they should render in mobile or desktop mode.
+ *
+ * @returns `true` if mobile, `false` otherwise
+ */
```
Comment (JoCa96) [inline packages/sit-onyx/src/components/OnyxNavItem/future/types.ts:14]: Please update the copied comment, to correctly reflect the purpose of this injection key

File: packages/sit-onyx/src/components/OnyxNavItem/future/types.ts [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+import type { InjectionKey } from "vue";
+
+export type OnyxNavItemProps = {
+  href?: string;
+  active?: boolean;
+};
+
+/**
+ * [Vue injection key](https://vuejs.org/guide/components/provide-inject) that is provided by the flyout menu
+ * to provide child components with correct aria-patterns.
```
Comment (JoCa96) [inline packages/sit-onyx/src/components/OnyxNavItem/future/types.ts:11]:
```suggestion
 * to provide child components with correct aria attributes.
```

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/components/OnyxNavItem/future/OnyxNavItem.vue [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+}>();
+
+const menuButton = inject(MENU_BUTTON_ITEM_INJECTION_KEY);
+</script>
+<template>
+  <OnyxListItem :selected="props.active" v-bind="menuButton?.listItem" class="onyx-future-nav-item">
+    <a
+      class="onyx-future-nav-item__anchor"
+      v-bind="menuButton?.menuItem({ active: props.active })"
+      :href="props.href ?? 'javascript:void(0)'"
```
[the remark this answers is not in the document]
Comment (MajaZarkova) [reply packages/sit-onyx/src/components/OnyxNavItem/future/OnyxNavItem.vue:21]: If there is no href, the element is not focusable. This is why Jonathan added this

File: packages/sit-onyx/src/components/OnyxNavItem/future/types.ts [hunk 12]
```diff
@@ -0,0 +1,17 @@
+import type { createMenuButton } from "@sit-onyx/headless";
+import type { InjectionKey } from "vue";
+
+export type OnyxNavItemProps = {
+  href?: string;
+  active?: boolean;
+};
```
[the remark this answers is not in the document]
Comment (MajaZarkova) [reply packages/sit-onyx/src/components/OnyxNavItem/future/types.ts:7]: sure

**Review by larsrickert — APPROVED** [review]

**Pull request merged**